### PR TITLE
support setting the default schema in the SQLRequest

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: sql/src/main/java/io/crate/analyze/Analysis.java [hunk 1]
```diff
@@ -23,9 +23,12 @@
 
 import io.crate.analyze.relations.AnalyzedRelation;
 
+import javax.annotation.Nullable;
```
Comment (mfelsche) [inline sql/src/main/java/io/crate/analyze/Analysis.java]: unused?

File: sql/src/test/java/io/crate/analyze/DeleteAnalyzerTest.java [hunk 2]
```diff
@@ -158,6 +160,4 @@ public void testDeleteWhereVersionIsNullPredicate() throws Exception {
         analyze("delete from users where _version is null",
                 new Object[]{1});
     }
-
```
Comment (mfelsche) [inline sql/src/test/java/io/crate/analyze/DeleteAnalyzerTest.java]: pls revert changes

File: sql/src/test/java/io/crate/integrationtests/DefaultSchemaIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    @Test
+    public void testSelectFromFooSchemaWithRequestHeaders() throws Exception {
+        execute(requestWithSchema("foo", "create table foobar (x string) with (number_of_replicas = 0)"));
+        ensureYellow();
+        waitNoPendingTasksOnAll();
+        execute(requestWithSchema("foo", "alter table foobar set (number_of_replicas = '0-1')"));
+
+        assertThat(getTableCount("foo", "foobar"), is(1L));
```
Comment (mfelsche) [inline sql/src/test/java/io/crate/integrationtests/DefaultSchemaIntegrationTest.java]: ensure that no table doc.foobar has been created.

**Comment by mfelsche** [comment]
ack

**Comment by mfussenegger** [comment]
merged to default-schema branch in https://github.com/crate/crate/pull/1853

**Pull request closed without merging**